modules/platforms/python/pyignite/cache.py

```diff
@@ -0,0 +1,592 @@
+        :return: old value or None.
+        """
+        result = cache_get_and_put_if_absent(
+            self._client, self._cache_id, key, value, key_hint, value_hint
+        )
+        result.value = self._process_binary(result.value)
+        return result
+
+    @status_to_exception(CacheError)
+    def put_if_absent(self, key, value, key_hint=None, value_hint=None):
+        """
+        Puts a value with a given key to cache only if the key does not
+        already exist.
+
+        :param key: key for the cache entry. Can be of any supported type,
+        :param value: value for the key,
+        :param key_hint: (optional) Ignite data type, for which the given key
+         should be converted,
+        :param value_hint: (optional) Ignite data type, for which the given
+         value should be converted.
+        """
+        return cache_put_if_absent(
+            self._client, self._cache_id, key, value, key_hint, value_hint
+        )
+
+    @status_to_exception(CacheError)
+    def get_and_remove(self, key, key_hint=None) -> Any:
+        """
+        Removes the cache entry with specified key, returning the value.
+
+        :param key: key for the cache entry. Can be of any supported type,
+        :param key_hint: (optional) Ignite data type, for which the given key
+         should be converted,
+        :return: old value or None.
+        """
+        result = cache_get_and_remove(
+            self._client, self._cache_id, key, key_hint
+        )
+        result.value = self._process_binary(result.value)
+        return result
+
+    @status_to_exception(CacheError)
+    def get_and_replace(
+        self, key, value, key_hint=None, value_hint=None
+    ) -> Any:
+        """
+        Puts a value with a given key to cache, returning previous value
+        for that key, if and only if there is a value currently mapped
+        for that key.
+
+        :param key: key for the cache entry. Can be of any supported type,
+        :param value: value for the key,
+        :param key_hint: (optional) Ignite data type, for which the given key
+         should be converted,
+        :param value_hint: (optional) Ignite data type, for which the given
+         value should be converted.
+        :return: old value or None.
+        """
+        result = cache_get_and_replace(
+            self._client, self._cache_id, key, value, key_hint, value_hint
+        )
+        result.value = self._process_binary(result.value)
+        return result
+
+    @status_to_exception(CacheError)
+    def remove_key(self, key, key_hint=None):
+        """
+        Clears the cache key without notifying listeners or cache writers.
+
+        :param key:  key for the cache entry,
+        :param key_hint: (optional) Ignite data type, for which the given key
+         should be converted,
+        """
+        return cache_remove_key(self._client, self._cache_id, key, key_hint)
+
+    @status_to_exception(CacheError)
```

Comment:
remove methods may mislead user if it' familiar with Cache.java:

Here is current mapping: java method -> pyignite method 
remove(Key) -> remove_key()
remove(Key, Value) -> remove_if_equals()
removeAll([Keys]) -> remove()
removeAll() -> remove()

I think following mapping is more clear to end user:
remove(Key) -> remove()
remove(Key, Value) -> remove_if_equals()
removeAll([Keys]) -> remove_all()
removeAll() -> remove_all()

Reply:
Don't you think it is rather misleading to name a method `remove_all()`, and make it remove *not* all, but only selected keys?

I understand that such naming can have historical reasons. But, frankly, I do not expect that Java programmers familiar with Cache.java would use *Python* Ignite thin client. I think its primary audience are Python developers who are unaware of Cache.java.

Reply:
For me as a Java developer it is ok to see removeAll(Collection) in API :)

The question is in a dialog between java developers and python developers. I think it is better when they use common language. In this case method names and it's behaviour came from Java.
Maybe we can add `remove_keys()` and `remove_key()` like it is made for `contains_*()` methods

Current realisation for remove_* methods looks a bit confusing for me

Reply:
A bit too verbose, but leaves no uncertainties:
- `remove(Key)` → `remove_key()`
- `remove(Key, Value)` → `remove_if_equals()`
- `removeAll([Keys])` → `remove_keys()`
- `removeAll()` → `remove_all()`

Can we proceed with this?

Reply:
Yes, looks good